OpenRA.Mods.Common/Traits/Render/DrawLineToTarget.cs

```diff
@@ -62,65 +45,61 @@ void INotifySelected.Selected(Actor a)
 			lifetime = info.Delay;
 		}
 
+		void INotifySelected.Selected(Actor a)
+		{
+			ShowTargetLines(a);
+		}
+
 		IEnumerable<IRenderable> IRenderAboveShroudWhenSelected.RenderAboveShroud(Actor self, WorldRenderer wr)
 		{
-			var force = Game.GetModifierKeys().HasModifier(Modifiers.Alt);
-			if ((lifetime <= 0 || --lifetime <= 0) && !force)
+			if (self.Owner != self.World.LocalPlayer)
 				yield break;
 
-			if (!(force || Game.Settings.Game.DrawTargetLine))
+			// Shift is "force" too; players want to see the lines when in waypoint mode.
+			var force = Game.GetModifierKeys().HasModifier(Modifiers.Alt)
+			          || Game.GetModifierKeys().HasModifier(Modifiers.Shift);
+
+			if (--lifetime <= 0 && !force)
 				yield break;
 
-			if (targets == null || targets.Count == 0)
+			if (!(force || Game.Settings.Game.DrawTargetLine))
 				yield break;
 
-			foreach (var target in targets)
+			var prev = self.CenterPosition;
+			var a = self.CurrentActivity;
+			for (; a != null; a = a.NextActivity)
 			{
-				if (target.Type == TargetType.Invalid)
+				if (a.IsCanceling)
 					continue;
 
-				yield return new TargetLineRenderable(new[] { self.CenterPosition, target.CenterPosition }, c);
+				foreach (var n in a.TargetLineNodes(self))
+				{
+					if (n.Target.Type != TargetType.Invalid)
+					{
+						yield return new TargetLineRenderable(new[] { prev, n.Target.CenterPosition }, n.Color, 2, 3);
```

Comment:
Can we please expose the `LineWidth` and `MarkerWidth` on the `DrawLineToTargetInfo`?

Reply:
Done.